### ISSUE TITLE
Reading task: Finding and correcting syntax errors using recurrent neural networks

### ISSUE THREAD

**Issue by MadeInGer** [sagr4019/ResearchProject#48]
Please read the paper 'Finding and correcting syntax errors usingrecurrent neural networks' as mentioned at [posted here](https://github.com/sagr4019/ResearchProject/issues/43#issuecomment-394182204)

**Comment by jan-christiansen** [comment]
Download data at https://archive.org/download/sensibility-replication-package and train network.

**Comment by jan-christiansen** [comment]
List run rimes for training.

**Comment by jan-christiansen** [comment]
What is the current state of this issue?

**Comment by DanielEggelmann** [comment]
This error can be resolved by replacing in "condensed_corpus.py"
`with io.BytesIO(blob) as filelike:`
        `return np.load(filelike)`

with
` return np.frombuffer(blob, dtype=np.uint8) `

**Comment by DanielEggelmann** [comment]
Setting `pickle_safe=False` in "train.py" for the function call of "model.fit_generator" gets it working:

- Very slow speed, 1000+ hours per generation (on gpu, batch_size seems to be 1)
- Implementation seems to be written for an older version of keras (1.11?), there might be some issues with changes during keras development since 1.11

**Comment by DanielEggelmann** [comment]
Training aborted after around 30 minutes:
`17599/91759831 [..............................] - ETA: 1660:44:25 - loss: 0.7643 - categorical_accuracy: 0.7565`

**Comment by DanielEggelmann** [comment]
Adjusting line 78 in "train.py" to:
`steps_per_epoch=training_batches.samples_per_epoch // BATCH_SIZE`
cuts estimated training time down to around 3 and a half hours.

**Comment by DanielEggelmann** [comment]
Second training attempt aborted after around 2 and a half hours:
` 131332/179218 [====================>.........] - ETA: 52:18 - loss: 0.7789 - categorical_accuracy: 0.7455 `
The accuracy seems to converge on 0.7.
